Modules/Core/TestKernel/include/itkTestingMacros.h

```diff
@@ -215,4 +215,14 @@ CLANG_PRAGMA_POP    \
     return EXIT_FAILURE; \
   } \
   object->Set##variable( value );
+
+/** The name of the executable, argv[0], is not always available as argv[0] may be null.
+ * In that case, use the name of the test function.
+*/
+#define itkNameOfTestExecutableMacro(argv) [argv](const std::string& functionName) \
```

Comment:
Convenient usage pattern results in this seriously cryptic syntax of the macro's definition 😄

Reply:
@dzenan Honestly I agree about the macro definition being cryptical. The implementation could very much be simplified if you guys think it's unnecessary to produce something like `"<" + __func__ + " executable>"`

You see, I wanted to produce a nice and meaningful string, specifying the name of the test executable, even when `argv[0]` is null. But now I feel a bit sorry about the cryptic syntax!

Reply:
That information is very useful, and the way to go. I just can believe that this compiles :smile:

Reply:
When you do `std::cout << itkNameOfTestExecutableMacro(argv);` a temporary function object is created by the lambda expression, and its compiler-generated `operator()` is called with `__func__` as argument. The `operator()` function call returns a string, specifying the name of the test executable. Hope it's clear :grinning: